### PR TITLE
Make libdawn_native a shared library.

### PR DESCRIPTION
**Pull request by Kangz**
PTAL. This is a bit more complicated than I expected because it forces us to sort-out our dependencies. (previously they leaked between binaries and dawn_native, making it compile when deps weren't set up properly).

### CODE REVIEW

File: src/dawn_native/dawn_platform.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// limitations under the License.
+
+#ifndef DAWNNATIVE_DAWNPLATFORM_H_
+#define DAWNNATIVE_DAWNPLATFORM_H_
+
+// Use dawncpp to have the enum and bitfield definitions
+#include <dawn/dawncpp.h>
+
+// Use our autogenerated version of the dawn structures that point to dawn_native object types
+// (nxt::Buffer is dawn_native::BufferBase*)
```
Comment (SenorBlanco) [inline src/dawn_native/dawn_platform.h]: s/nxt::Buffer/dawn::Buffer/ ?

File: third_party/CMakeLists.txt [hunk 2]
```diff
@@ -138,5 +140,8 @@ set(STB_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/stb PARENT_SCOPE)
 # glm matrix math library
 set(GLM_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/glm PARENT_SCOPE)
 
+# glm matrix math library
```
Comment (kainino0x) [inline third_party/CMakeLists.txt]: wrong comment here

File: src/common/vulkan_platform.h [hunk 3]
```diff
@@ -59,26 +61,62 @@ class VkNonDispatchableHandle {
         return mHandle != 0;
     }
 
-    static VkNonDispatchableHandle<Tag> CreateFromHandle(uint64_t handle) {
+    // The regular Vulkan handle type depends on the pointer width but is always 64 bits wide.
```
Comment (kainino0x) [inline src/common/vulkan_platform.h]: This reads as "The regular Vulkan handle type depends on the pointer width" + "The regular Vulkan handle type is always 64 bits wide" - sounds like a contradiction.

**Comment by kainino0x** [comment]
Also needs reformat

File: src/dawn_native/dawn_platform.h (hunk 1 above)
Comment (Kangz) [reply src/dawn_native/dawn_platform.h]: Done

File: third_party/CMakeLists.txt (hunk 2 above)
Comment (Kangz) [reply third_party/CMakeLists.txt]: Done

File: src/common/vulkan_platform.h (hunk 3 above)
Comment (Kangz) [reply src/common/vulkan_platform.h]: Done, explained better what this means.

**Comment by Kangz** [comment]
We don't have a good description of what the different targets are, I'll rework the README after the shared lib work.

**Pull request merged**